### PR TITLE
Fix multiple argument run/passthru escaping and migrate most to use it

### PR DESCRIPTION
**Pull request by andytson-inviqa**
* Multiple argument run/passthru was splitting arguments with spaces in, and losing empty arguments
* Some still use single argument run/passthru due to use of shell, but may be dealt with later

### CODE REVIEW

File: src/_base/docker/image/console/root/lib/sidekick.sh [hunk 1]
```diff
@@ -62,29 +81,50 @@ run()
...
+    fi
 
-    echo -e "\\033[${INDICATOR_PASSTHRU}■\\033[0m > $*"
+    prompt
 
-    if ! bash -c "$@"; then
-        exit 1
+    if [ "${DEPRECATED_MODE}" = "yes" ]; then
+        echo -e "\\033[${INDICATOR_PASSTHRU}■\\033[0m > $*" >&2
+        if ! bash -e -c "${COMMAND_DEPRECATED[@]}" >&2; then
```
Comment (andytson-inviqa) [inline src/_base/docker/image/console/root/lib/sidekick.sh]:
```suggestion
        if ! bash -e -c "${COMMAND_DEPRECATED[@]}"; then
```

File: src/spryker/harness.yml [hunk 2]
```diff
@@ -44,21 +44,21 @@ attributes:
...
-        - task "spryker:build"
+        - task spryker:build
     install:
       steps:
-        - task "rabbitmq:vhosts"
-        - task "http:wait" $JENKINS_URL
-        - task "spryker:install"
-        - task "overlay:apply"
+        - task rabbitmq:vhosts
+        - task http:wait' $JENKINS_URL
```
Comment (andytson-inviqa) [inline src/spryker/harness.yml]:
```suggestion
        - task http:wait $JENKINS_URL
```

File: src/magento2/harness.yml [hunk 3]
```diff
@@ -61,60 +61,60 @@ attributes:
...
+        - run rm -f /app/app/etc/env.php
+        - run rm -f /app/app/etc/config.php
         - |
           passthru "magento setup:install \
-            --key=${MAGENTO_CRYPT_KEY} \
+            --key='${MAGENTO_CRYPT_KEY}' \
             \
             --backend-frontname=admin \
-            --base-url=https://${APP_HOST}/ \
+            --base-url='https://${APP_HOST}/'' \
```
Comment (kierenevans) [inline src/magento2/harness.yml:92]:
```suggestion
            --base-url='https://${APP_HOST}/' \
```

File: src/spryker/docker/image/console/root/lib/task/skeleton/apply.sh.twig [hunk 4]
```diff
@@ -6,15 +6,15 @@ function task_skeleton_apply()
...
 
 function fetch_demoshop()
 {
-    run "git clone --depth 1 --branch 201903.0-p1 {{ @('spryker.demoshop-url') }} /tmp/spryker"
+    run git clone --depth 1 --branch 201903.0-p1 "{{ @('spryker.demoshop-url') }}" /tmp/spryker
     run "shopt -s dotglob && mv /tmp/spryker/* /app"
-    run "rm -rf /tmp/spryker"
-    run "rm -rf /app/.git"
+    run rm -rf /tmp/spryker"
+    run rm -rf /app/.git"
```
Comment (kierenevans) [inline src/spryker/docker/image/console/root/lib/task/skeleton/apply.sh.twig]:
```suggestion
    run rm -rf /tmp/spryker
    run rm -rf /app/.git
```

File: src/magento2/harness.yml (hunk 3 above)
Comment (kierenevans) [inline src/magento2/harness.yml]:
```suggestion
            --base-url='https://${APP_HOST}/' \
```

File: src/spryker/docker/image/console/root/lib/task/spryker/install.sh [hunk 5]
```diff
@@ -2,5 +2,5 @@
 
 function task_spryker_install()
 {
-    passthru "PGPASSWORD=$DB_PASS vendor/bin/install -r docker -x frontend"
+    PGPASSWORD=$DB_PASS passthru vendor/bin/install -r docker -x frontend
```
Comment (kierenevans) [inline src/spryker/docker/image/console/root/lib/task/spryker/install.sh]:
```suggestion
    PGPASSWORD="$DB_PASS" passthru vendor/bin/install -r docker -x frontend
```

File: src/spryker/harness.yml [hunk 6]
```diff
@@ -44,21 +44,21 @@ attributes:
...
-        - task "spryker:build"
+        - task spryker:build
     install:
       steps:
-        - task "rabbitmq:vhosts"
-        - task "http:wait" $JENKINS_URL
-        - task "spryker:install"
-        - task "overlay:apply"
+        - task rabbitmq:vhosts
+        - task http:wait $JENKINS_URL
```
Comment (kierenevans) [inline src/spryker/harness.yml]:
```suggestion
        - task http:wait "$JENKINS_URL"
```

File: src/spryker/harness.yml [hunk 7]
```diff
@@ -44,21 +44,21 @@ attributes:
...
+        - task spryker:install
+        - task overlay:apply
     init:
       steps:
-        - task "http:wait" $JENKINS_URL
-        - task "rabbitmq:vhosts"
-        - task "spryker:init"
+        - task http:wait $JENKINS_URL
```
Comment (kierenevans) [inline src/spryker/harness.yml]:
```suggestion
        - task http:wait "$JENKINS_URL"
```

File: src/spryker/harness.yml [hunk 8]
```diff
@@ -181,13 +181,13 @@ command('frontend yves watch'):
     COMPOSE_PROJECT_NAME: = @('namespace')
   exec: |
     #!bash(workspace:/)|@
-    passthru "docker-compose exec -u build console bash -i -c 'cd @('frontend.path'); @('frontend.yves.watch')'"
+    passthru docker-compose exec -u build --workdir '@('frontend.path')' console @('frontend.yves.watch')
```
Comment (kierenevans) [inline src/spryker/harness.yml:184]: :sunglasses:

File: src/drupal8/harness.yml [hunk 9]
```diff
@@ -35,35 +35,35 @@ attributes:
...
+              "--db-user=${DB_USER}" \
+              "--db-pass=${DB_PASS}" \
+              "--db-port=3306" \
               \
               --site-name=Drupal \
               --site-mail=admin@example.com \
               \
               --account-name=admin \
               --account-mail=admin@example.com \
               --account-pass=admin123"
```
Comment (kierenevans) [inline src/drupal8/harness.yml]:
```suggestion
              --account-pass=admin123
```

File: src/magento1/harness.yml [hunk 10]
```diff
@@ -52,48 +52,48 @@ attributes:
...
+            --db_model mysql4 \
+            --db_host "${DB_HOST}" \
+            --db_name "${DB_NAME}" \
+            --db_user "${DB_USER}" \
+            --db_pass "${DB_PASS}" \
+            --url "https://${APP_HOST}/" \
+            --skip_url_validation yes \
+            --use_rewrites yes \
+            --use_secure yes \
+            --secure_base_url "https://${APP_HOST}/"" \
```
Comment (kierenevans) [inline src/magento1/harness.yml]:
```suggestion
            --secure_base_url "https://${APP_HOST}/" \
```

File: src/magento1/harness.yml [hunk 11]
```diff
@@ -52,48 +52,48 @@ attributes:
...
-            --session_save 'db' \
-            --admin_frontname 'admin'"
-        - run "rm -f /app/public/app/etc/local.xml"
-        - task "overlay:apply"
-        - task "assets:dump"
+            --admin_username admin \
+            --admin_password admin123admin123 \
+            --encryption_key "${MAGENTO_CRYPT_KEY}" \
+            --session_save db \
+            --admin_frontname admin"
```
Comment (kierenevans) [inline src/magento1/harness.yml]:
```suggestion
            --admin_frontname admin
```

File: src/_base/harness/config/events.yml [hunk 12]
```diff
@@ -3,10 +3,13 @@ after('harness.install'): |
...
 
-after('harness.refresh'): |
-  #!bash(workspace:/)|@
-  run docker-compose -p @('namespace') stop
-  passthru "docker-compose config --services | grep -v php-fpm | xargs docker-compose pull"
-  passthru "docker-compose config --services | grep -v cron | grep -v job-queue-consumer | grep -v jenkins-runner | xargs docker-compose build --pull"
-  passthru "docker-compose config --services | grep -v cron | grep -v job-queue-consumer | grep -v jenkins-runner | xargs docker-compose up -d"
-  run docker-compose -p @('namespace') up -d --build
+after('harness.refresh'):
+  env:
```
Comment (kierenevans) [inline src/_base/harness/config/events.yml:7]: Not sure this works for events

Comment (andytson-inviqa) [reply src/_base/harness/config/events.yml:7]: buggy in the first place as well as the passthru's didn't set -p

Comment (kierenevans) [reply src/_base/harness/config/events.yml:7]: True!

Comment (andytson-inviqa) [reply src/_base/harness/config/events.yml:7]: David says it works for these

File: src/magento2/harness.yml (hunk 3 above)
Comment (andytson-inviqa) [inline src/magento2/harness.yml]:
```suggestion
            --base-url='https://${APP_HOST}/' \
```

File: src/magento1/harness.yml [hunk 13]
```diff
@@ -52,48 +52,48 @@ attributes:
...
+            --db_host "${DB_HOST}" \
+            --db_name "${DB_NAME}" \
+            --db_user "${DB_USER}" \
+            --db_pass "${DB_PASS}" \
+            --url "https://${APP_HOST}/" \
+            --skip_url_validation yes \
+            --use_rewrites yes \
+            --use_secure yes \
+            --secure_base_url "https://${APP_HOST}/" \
+            --use_secure_admin yes' \
```
Comment (kierenevans) [inline src/magento1/harness.yml]:
```suggestion
            --use_secure_admin yes \
```

File: src/wordpress/harness.yml [hunk 14]
```diff
@@ -39,27 +39,27 @@ attributes:
...
     init:
       steps:
-        - run "bin/wp-cli.phar --path='${WORDPRESS_INSTALL_DIRECTORY}' option update siteurl $(eval echo $WORDPRESS_URL)"
-        - run "bin/wp-cli.phar --path='${WORDPRESS_INSTALL_DIRECTORY}' option update home $(eval echo $WORDPRESS_URL)"
-        - run "bin/wp-cli.phar --path='${WORDPRESS_INSTALL_DIRECTORY}' user create admin admin@localhost.local --role=administrator --user_pass=admin123 || exit 0"
-        - run "bin/wp-cli.phar --path='${WORDPRESS_INSTALL_DIRECTORY}' user update admin --role=administrator --user_pass=admin123 --skip-email"
+        - run bin/wp-cli.phar "--path=${WORDPRESS_INSTALL_DIRECTORY}" option update siteurl "$(eval echo "${WORDPRESS_URL}")"
+        - run bin/wp-cli.phar "--path=${WORDPRESS_INSTALL_DIRECTORY}" option update home "$(eval echo "${WORDPRESS_URL}")"
+        - run "bin/wp-cli.phar '--path=${WORDPRESS_INSTALL_DIRECTORY}' user create admin admin@localhost.local --role=administrator --user_pass=admin123 || exit 0"
+        - run bin/wp-cli.phar "--path=${WORDPRESS_INSTALL_DIRECTORY}" user update admin --role=administrator --user_pass=admin123 --skip-email"
```
Comment (kierenevans) [inline src/wordpress/harness.yml]:
```suggestion
        - run bin/wp-cli.phar "--path=${WORDPRESS_INSTALL_DIRECTORY}" user update admin --role=administrator --user_pass=admin123 --skip-email
```

**Review by kierenevans — APPROVED** [review]

**Pull request merged**